src/app/payment/paymentMethod/StripePaymentMethod.tsx

```diff
@@ -3,65 +3,102 @@ import React, { useCallback, FunctionComponent } from 'react';
-import { TranslatedString } from '../../locale';
+import {  TranslatedString } from '../../locale';
 
 import HostedWidgetPaymentMethod, { HostedWidgetPaymentMethodProps } from './HostedWidgetPaymentMethod';
+import StripeV3CustomCardForm from './StripeV3CustomCardForm';
 
 export type StripePaymentMethodProps = Omit<HostedWidgetPaymentMethodProps, 'containerId'>;
 
 export interface StripeOptions {
     alipay?: StripeElementOptions;
     card: StripeElementOptions;
+    cardCvc: StripeElementOptions;
+    cardExpiry: StripeElementOptions;
+    cardNumber: StripeElementOptions;
     iban: StripeElementOptions;
     idealBank: StripeElementOptions;
 }
 interface WithCheckoutStripePaymentMethodProps {
     storeUrl: string;
 }
-
-export enum StripeV3PaymentMethodType {
+export enum StripeElementType {
     alipay = 'alipay',
     card = 'card',
+    cardCvc = 'cardCvc',
+    cardExpiry = 'cardExpiry',
+    cardNumber = 'cardNumber',
     iban = 'iban',
     idealBank = 'idealBank',
 }
-
 const StripePaymentMethod: FunctionComponent<StripePaymentMethodProps & WithCheckoutStripePaymentMethodProps> = ({
       initializePayment,
       method,
       storeUrl,
       ...rest
   }) => {
-    const paymentMethodType = method.id as StripeV3PaymentMethodType;
-    const additionalStripeV3Classes = paymentMethodType !== StripeV3PaymentMethodType.alipay ? 'optimizedCheckout-form-input widget--stripev3' : '';
+    const { useIndividualCardFields } = method.initializationData;
+    const paymentMethodType = method.id as StripeElementType;
+    const additionalStripeV3Classes = paymentMethodType !== StripeElementType.alipay ? 'optimizedCheckout-form-input widget--stripev3' : '';
     const containerId = `stripe-${paymentMethodType}-component-field`;
+    const classes = {
+        base: 'form-input optimizedCheckout-form-input',
+    };
+    const stripeOptions: StripeOptions = {
+        [StripeElementType.card]: {
+            classes,
+        },
+        [StripeElementType.cardCvc]: {
+            classes,
+            placeholder: '',
+        },
+        [StripeElementType.cardExpiry]: {
+            classes,
+        },
+        [StripeElementType.cardNumber]: {
+            classes,
+            showIcon: true,
+            placeholder: '',
+        },
+        [StripeElementType.iban]: {
+            ...{ classes },
+            supportedCountries: ['SEPA'],
+        },
+        [StripeElementType.idealBank]: {
+            classes,
+        },
+    };
 
-    const initializeStripePayment = useCallback(async (options: PaymentInitializeOptions) => {
-        const classes = {
-            base: 'form-input optimizedCheckout-form-input',
+    const getIndividualCardElementOptions = useCallback((stripeInitializeOptions: StripeOptions) => {
+        return {
+            cardNumberElementOptions: { containerId: 'stripe-card-number-component-field', ...stripeInitializeOptions[StripeElementType.cardNumber] },
```

Comment:
for readability, would be good to format this in different lines
```js
cardNumberElementOptions: {
  containerId: 'stripe-card-number-component-field', 
  ...stripeInitializeOptions[StripeElementType.cardNumber]
},
cardExpiryElementOptions: {
  containerId: 'stripe-expiry-component-field',
  ...stripeInitializeOptions[StripeElementType.cardExpiry]
},
// ...
```

Whenever a line exceeds 120 chars, consider breaking it into different lines